futures-util/src/stream/buffer_unordered.rs

```diff
@@ -134,6 +134,16 @@ where
     }
 }
 
+impl<St> FusedStream for BufferUnordered<St>
+where
+    St: Stream,
+    St::Item: Future,
+{
+    fn is_terminated(&self) -> bool {
+        self.in_progress_queue.is_empty() && self.stream.is_terminated()
```

Comment:
This can return `true` before `poll` returns a `None`. Is that valid for `FusedStream`, or does this need to track whether it's returned a `None` before it can return `true`?